app/src/main/java/com/github/orkest/bluetooth/data/BluetoothServiceManager.kt

```diff
@@ -2,27 +2,37 @@ package com.github.orkest.bluetooth.data
 
 
 import android.Manifest
-import android.bluetooth.BluetoothAdapter
-import android.bluetooth.BluetoothDevice
-import android.bluetooth.BluetoothManager
+import android.annotation.SuppressLint
+import android.bluetooth.*
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
 import android.content.IntentFilter
 import android.content.pm.PackageManager
+import android.os.Handler
+import android.os.Looper
 import android.util.Log
 import androidx.activity.result.ActivityResultLauncher
+import androidx.annotation.RequiresPermission
 import androidx.core.app.ActivityCompat
-import com.github.orkest.bluetooth.domain.BluetoothInterface
+import com.github.orkest.bluetooth.domain.*
+import com.github.orkest.bluetooth.domain.BluetoothConstants.Companion.MY_UUID
+import com.github.orkest.bluetooth.domain.BluetoothConstants.Companion.NAME
+import com.github.orkest.data.Constants
+import java.io.IOException
 
-class BluetoothServiceManager : BluetoothInterface {
+class BluetoothServiceManager(private var handler: Handler) : BluetoothInterface {
 
 
-    override val devices: MutableMap<String, String> = mutableMapOf()
+    override var devices: MutableList<Device> = mutableListOf()
```

Comment:
why did you switch to var?

Reply:
I wanted to be able to assign it directly below when checking for already paired devices